sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala

```diff
@@ -372,6 +372,58 @@ class JsonExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     )
   }
 
+  test("from_json - array") {
+    val schema = ArrayType(StructType(StructField("a", IntegerType) :: Nil))
+
+    // json array: `Array(Row(...), ...)`
+    val jsonData1 = """[{"a": 1}, {"a": 2}]"""
+    val expected =
+      InternalRow.fromSeq(1 :: Nil) ::
+      InternalRow.fromSeq(2 :: Nil) :: Nil
+    checkEvaluation(JsonToStruct(
+      schema, Map.empty, Literal(jsonData1), gmtId), expected)
+
+    // json object: `Array(Row(...))`
+    val jsonData2 = """{"a": 1}"""
```

Comment:
I would make each example a separate test. This way it's easier to figure out what breaks later.

e.g.
  1. `from_json - input=array, schema=array, output=array`
  2. `from_json - input=object, schema=array, output=array of single object`
  3. `from_json - input=empty json array, schema=array, output=empty array`
...